src/endpoints/user.py

```diff
@@ -0,0 +1,298 @@
+    verify_password,  # isort: skip
+)  # isort: skip
+
+router = APIRouter(
+    prefix="/user", tags=["user"], responses={401: {"user": "Not authorized"}}
+)
+
+
+@router.get("", status_code=status.HTTP_200_OK, response_model=None)
+async def filter_user(
+    email: str | None = None,
+    username: str | None = None,
+    first_name: str | None = None,
+    last_name: str | None = None,
+    contact_number: str | None = None,
+    address: str | None = None,
+    librarian: dict = Depends(get_current_librarian),
+    db: Session = Depends(get_db),
+) -> None:
+    """
+    Filters the users list based on param provided. If None given then it will
+    return the complete list.
+    """
+    params = {
+        "email": email,
+        "username": username,
+        "first_name": first_name,
+        "last_name": last_name,
+        "contact_number": contact_number,
+        "address": address,
+    }
+    filters = {key: value for key, value in params.items() if value}
+    try:
+        users = db.query(User).filter_by(**filters).all()
+        return users
+    except Exception:
+        logging.exception(f"Exception occured -- {__name__}.filter_user")
+        raise db_not_available()
+
+
+@router.get("/", status_code=status.HTTP_200_OK, response_model=None)
+async def get_all_users(
+    librarian: dict = Depends(get_current_librarian), db: Session = Depends(get_db)
+) -> List[User]:
+    """
+    Returns the list of all the users registered in a library.
+    Param
+    -----
+    JWT token of a librarian.
+    Throws an exception if JWT is not of librarian\n
+    Returns
+    ------
+    List of users: List[User]
+    """
+    try:
+        users = db.execute(select(User).where(True)).scalars().all()
+        logging.info(f"Returning all users. -- {__name__}.get_all_user")
+        return users
+    except Exception:
+        logging.exception(f"Exception occured -- {__name__}.get_all_user")
+        raise db_not_available()
+
+
+@router.get("/{user_id}", status_code=status.HTTP_200_OK, response_model=None)
+async def get_user_by_id(
+    librarian: dict = Depends(get_current_librarian),
+    db: Session = Depends(get_db),
+    user_id: int = Path(gt=0),
+) -> User:
+    """
+    Returns a single user.\n
+    Param
+    -----
+    user_id: int\n
+    JWT token of a librarian.
+    Throws an exception if JWT is not of librarian\n
+    Returns
+    ------
+    user: User
+    """
+    try:
+        user = db.execute(select(User).where(User.id == user_id)).scalars().first()
+    except Exception:
+        logging.exception(f"Exception occured -- {__name__}.get_user_by_id")
+        raise db_not_available()
+    if user:
+        logging.info(f"Returning a single user. -- {__name__}.get_user_by_id")
+        return user
+    else:
+        logging.error(f"User not found -- {__name__}.get_user_by_id")
+        raise user_not_exist()
+
+
+@router.delete("/delete_user", status_code=status.HTTP_204_NO_CONTENT)
```

Comment:
I think we can skip the "delete_user" and "update_user" keywords since the methods DELETE and PUT are self-explanatory in what they do. Normally API endpoints don't have these keywords.

Reply:
Ohkay ! will remove those.